notebook/notebookapp.py

```diff
@@ -704,6 +707,44 @@ class NotebookApp(JupyterApp):
 
     _log_formatter_cls = LogFormatter
 
+    _json_logging_import_error_logged = False
+
+    log_json = Bool(False, config=True,
+        help=_('Set to True to enable JSON formatted logs. '
+               'Run "pip install notebook[json-logging]" to install the '
+               'required dependent packages. Can also be set using the '
+               'environment variable JUPYTER_ENABLE_JSON_LOGGING=true.')
+    )
+
+    @default('log_json')
+    def _default_log_json(self):
+        """Get the log_json value from the environment."""
+        return os.getenv('JUPYTER_ENABLE_JSON_LOGGING', 'false').lower() == 'true'
+
+    @validate('log_json')
+    def _validate_log_json(self, proposal):
+        # If log_json=True, see if the json_logging package can be imported and
+        # override _log_formatter_cls if so.
+        value = proposal['value']
+        if value:
+            try:
+                import json_logging
+                self._log_formatter_cls = json_logging.JSONLogFormatter
+                # Note that we don't need to trigger _log_format_changed here
+                # because init_logging will set the JSONLogFormatter on all
+                # of the registered loggers.
+            except ImportError:
+                # If configured for json logs and we can't do it, log a hint.
+                # Only log the error once though.
+                if not self._json_logging_import_error_logged:
+                    logging.getLogger(__name__).exception(
```

Comment:
I think we should just log a warning here rather than including the exception.  This causes unnecessary noise that the message addresses anyway:
```
Unable to use json logging due to missing packages. Run "pip install notebook[json-logging]" to fix.
Traceback (most recent call last):
  File "/opt/anaconda3/envs/notebook-dev/lib/python3.7/site-packages/notebook/notebookapp.py", line 731, in _validate_log_json
    import json_logging
ModuleNotFoundError: No module named 'json_logging'
```

Reply:
Done.